source/common/router/config_impl.h

```diff
@@ -381,6 +381,43 @@ class RouteTracingImpl : public RouteTracing {
   Tracing::CustomTagMap custom_tags_;
 };
 
+/**
+ * Implementation of InternalRedirectPolicy that reads from the proto
+ * InternalRedirectPolicy of the RouteAction.
+ */
+class InternalRedirectPolicyImpl : public InternalRedirectPolicy {
+public:
+  explicit InternalRedirectPolicyImpl(
+      const envoy::config::route::v3::InternalRedirectPolicy& policy_config,
+      ProtobufMessage::ValidationVisitor& validator, absl::string_view current_route_name);
+  InternalRedirectPolicyImpl() = default;
+
+  bool enabled() const override { return enabled_; }
+
+  bool shouldRedirectForCode(const Http::Code& response_code) const override {
+    return redirect_response_codes_.contains(response_code);
+  }
+
+  std::vector<InternalRedirectPredicateSharedPtr> predicates() const override;
+
+  uint32_t maxInternalRedirects() const override { return max_internal_redirects_; }
+
+  bool isCrossSchemeRedirectAllowed() const override { return allow_cross_scheme_redirect_; }
+
+private:
+  absl::flat_hash_set<Http::Code> buildRedirectResponseCodes(
+      const envoy::config::route::v3::InternalRedirectPolicy& policy_config) const;
+
+  const bool enabled_{false};
```

Comment:
super nit - order bools at the bottom for packing?

Reply:
done.